inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/config/model/instrumentation/InstrumentationSettings.java

```diff
@@ -21,31 +23,41 @@
      * The configuration of internal parameters regarding the instrumentation process.
      */
     @Valid
+    @NotNull
     private InternalSettings internal;
 
     /**
      * The configuration for all special sensors.
      */
     @Valid
+    @NotNull
     private SpecialSensorSettings special;
 
     /**
      * Defines which packages of the bootstrap should not be instrumented.
      * All classes from the given packages and their subpackages will be ignored.
      */
-    private Map<String, Boolean> ignoredBootstrapPackages;
+    @NotNull
+    private Map<@NotBlank String, Boolean> ignoredBootstrapPackages;
+
+    /**
+     * All defined custom data providers, the key defines their name.
+     * The name is case sensitive!
+     */
+    @NotNull
+    private Map<@NotBlank String, @Valid GenericDataProviderSettings> dataProviders = Collections.emptyMap();
 
     /**
      * The configuration of the defined scopes. The map's key represents an unique id for the related instrumentation scope.
      */
     @Valid
     @NotNull
-    private Map<String, InstrumentationScopeSettings> scopes = Collections.emptyMap();
+    private Map<@NotBlank String, InstrumentationScopeSettings> scopes = Collections.emptyMap();
```

Comment:
would we also need a Map<@NotBlank String, @Valid InstrumentationScopeSettings>to make sure that values are valid? same below?

Reply:
We already have valid as an annotation on the attirbute there, this propagates to the type parameter. But iI guess it is more readable to have it explicitly there.